packages/sdks/react/src/lib/useFlag.ts

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useState, useRef } from 'react';
+import {
+  AllowedVariableType,
+  ChangeTypes,
+  logger,
+} from '@ninetailed/experience.js-shared';
+import { useNinetailed } from './useNinetailed';
+import { ChangesState } from '@ninetailed/experience.js';
+
+export type ChangeStatus = 'loading' | 'success' | 'error';
+
+export interface ChangeResult<T> {
+  value: T;
+  isLoading: boolean;
+  status: ChangeStatus;
+  error: Error | null;
+}
+
+/**
+ * Custom hook to retrieve a specific feature flag from Ninetailed changes.
+ *
+ * @param flagKey - The key of the feature flag to retrieve
+ * @param defaultValue - The default value to use if the flag is not found
+ * @returns An object containing the flag value and metadata about the request
+ */
+export function useFlag<T extends AllowedVariableType>(
+  flagKey: string,
+  defaultValue: T
+): ChangeResult<T> {
+  const ninetailed = useNinetailed();
+
+  const lastProcessedState = useRef<ChangesState | null>(null);
+
+  const [result, setResult] = useState<ChangeResult<T>>({
+    value: defaultValue,
+    isLoading: true,
+    status: 'loading',
+    error: null,
+  });
+
+  useEffect(() => {
+    const unsubscribe = ninetailed.onChangesChange((changesState) => {
+      if (
+        lastProcessedState.current &&
+        lastProcessedState.current.status === changesState.status &&
+        lastProcessedState.current.changes === changesState.changes &&
```

Comment:
Since changes is an array of objects, and technically we can't really assume that its identity is the same every time the changes are identical, I think we should make a deep comparison here.
If `changes` is also not an ordered list, then we should even do a set comparison here.

Probably for now a deep comparison should suffice though using `isEqual` from `radash`.

Then again, if you use that, you might as well compare the entire `lastProcessedState.current` to `changesState` using that.